clusterloader2/pkg/test/modules.go

```diff
@@ -36,8 +36,8 @@ import (
 //         -step 5
 // - step 6
 // the method will return [step1, step3, step5, step6].
-func flattenModuleSteps(ctx Context, unprocessedSteps []api.Step) ([]*api.Step, error) {
-	var processedSteps []*api.Step
+func flattenModuleSteps(ctx Context, unprocessedSteps []api.Step) ([]api.Step, error) {
```

Comment:
Let's not change this - passing the whole objects around is a performance hit.

Reply:
Updated, the ExecuteTest func signature will be slightly uglier now because we need to pass the slice of pointers down

Reply:
I wonder whether we can change the CL2 API to use slices of pointers everywhere instead of slices of structs. E.g. [here](https://github.com/kubernetes/perf-tests/blob/039c642e4d11d8c4c1a438f5e8bc2b56c1fc7eef/clusterloader2/api/types.go#L50) we would change
```
Steps []Step `json:"steps"`
```
to 
```
Steps []*Step `json:"steps"`
```

IIUC, it should be transparent to the users (no changes in the yaml files would be needed), serialization code would work without any changes and we wouldn't have to conversion from `[]A` to '[]*A` in the codebase. Thoughts?

Reply:
It'll make the Config struct less awkward to work with and I can make this PR a lot cleaner